### PR TITLE
feat: Add Spotlight component

### LINKED ISSUE #55: Create a 'Spotlight' component
We're looking to add 'Spotlight' component to motion-primitives! This component would let you highlight elements on a webpage dynamically, following the cursor.

### Requirements:
- Allow customization with transition and className.
- Maintain performance for smooth animations.
- Ensure compatibility with the most recent browsers.
- Simple, easy-to-use API.

### Ideal Contribution:
- A React component that uses Framer Motion for animations.
- Include a demo page (`docs/spotlight/page.mdx`) that demonstrates the spotlight effect on different types of content (e.g., text, images, buttons).

### Example Usage:
```jsx
<Spotlight
  transition={{
    type: 'spring',
    stiffness: 300,
    damping: 30,
  }}
  className="bg-zinc-50 size-14" 
>
  <div className="highlighted-element">
    <h2>Spotlight Feature</h2>
    <p>This area is dynamically highlighted by the spotlight component.</p>
  </div>
</Spotlight>
```

### How to Contribute:
- Fork the repository on GitHub.
- Create your feature branch (git checkout -b feature/spotlight).
- Commit your changes (git commit -am 'Add Spotlight component').
- Push to the branch (git push origin feature/spotlight).
- Open a new Pull Request targeting the main branch.

If you have a X handle, please provide it so I can tag you in the related announcement demo tweet. You'll also be added to the **Credit** in the documentation page. Thank you for considering contributing to motion-primitives!

### PR DESCRIPTION
**Pull request by divyanshu-vashishth**
### Description

Added a Spotlight component that highlights elements by following the cursor. It’s customizable with options for size, color, opacity, and transition. Examples for basic usage and card-based layouts are included.

### Questions:
- Are there any additional features or improvements needed?

resolved #55

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -21,6 +21,6 @@
       "@/*": ["./*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "app/docs/spotlight/page.mdx"],
```
Comment (ibelick) [inline tsconfig.json:24]: remove this please

File: components/core/spotlight.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { cn } from '@/lib/utils';
+
+type SpotlightProps = {
+  children: React.ReactNode;
+  className?: string;
+  spotlightSize?: number;
+  color?: string;
+  opacity?: number;
+  blur?: number;
+  transition?: {
```
Comment (ibelick) [inline components/core/spotlight.tsx:13]: you can use `SpringOptions` type from framer-motion. And rename the prop `springOptions`

**Comment by ibelick** [comment]
Thanks for the PR @divyanshu-vashishth!

It looks great, I'm thinking it could be even better if we tweak the API to be more like the [Cursor](https://github.com/ibelick/motion-primitives/blob/main/components/core/cursor.tsx). Imagine allowing the `Spotlight` to be used within any container, like this:

```tsx
<div className="card bg-zinc-500 rounded-md ...">
  <Spotlight className="bg-red-500 ..." />
  <p>Some text here</p>
</div>
```

Let me know your thougts on this!

**Comment by divyanshu-vashishth** [comment]
Thanks for the feedback!

- I've replaced the previous transition prop with springOptions, using SpringOptions from Framer Motion. This provides more flexibility in controlling the spotlight's spring behavior.
- The Spotlight can now be used directly inside any container, similar to the Cursor component. It works seamlessly within various parent components.

Let me know if there's anything else I am missing or doing wrong, you'd like me to tweak! 🚀

**Comment by ibelick** [comment]
@divyanshu-vashishth  
Thanks for the updates. Could you adjust the Spotlight to be used within a container? [Like the code here](https://github.com/ibelick/motion-primitives/pull/67#issuecomment-2408709544). Instead of a standalone Spotlight card, we want it integrated into existing containers. Also, could you simplify the examples to just two and make them a bit less fancy? The Spotlight with Maps is cool, but it doesn’t quite fit the library's vibe. Perhaps use something more basic like this: [Basic Card Example](https://motion-primitives-git-fork-divy-b08430-julienthibeauts-projects.vercel.app/docs/border-trail#border-trail-loading-card). Thanks again!

**Comment by ibelick** [comment]
Thanks for the PR @divyanshu-vashishth! Cf https://github.com/ibelick/motion-primitives/pull/78

**Pull request closed without merging**